conjure/objectwriter.go

```diff
@@ -45,12 +64,32 @@ func writeObjectType(file *jen.Group, objectDef *types.ObjectType) {
 				// double quotes instead.
 				fieldTags["conjure-docs"] = strings.Replace(strings.TrimSpace(string(fieldDef.Docs)), "`", `"`, -1)
 			}
+
+			// Add safety struct tag based on field's type safety (with recursive computation)
+			fieldSafety := getFieldSafety(fieldDef.Type)
+			if !fieldSafety.IsUnknown() {
+				switch fieldSafety.Value() {
+				case spec.LogSafety_SAFE:
+					fieldTags["safelogging"] = "@Safe"
+				case spec.LogSafety_UNSAFE:
+					fieldTags["safelogging"] = "@Unsafe"
+				case spec.LogSafety_DO_NOT_LOG:
+					fieldTags["safelogging"] = "@DoNotLog"
+				}
+			}
```

Comment:
This logic is basically duplicated here and in the above block. I think it makes sense to decomp a function that does the safety -> string representation translation ("@Safe", "@Unsafe", "@DoNotLog", or blank/nil) and then logic here and in the other place will share representation and can remove redundancy of assignment logic in different cases